crates/sui-protocol-config/src/lib.rs

```diff
@@ -454,6 +524,24 @@ impl ProtocolConfig {
         // To change the values here you must create a new protocol version with the new values!
         match version.0 {
             1 => Self {
+                max_tx_in_batch: Some(10),
+                max_modules_in_publish: Some(8),
+                max_arguments: Some(16),
+                max_type_arguments: Some(128),
+                max_type_argument_depth: Some(16),
+                // TODO: does this need to be big? Will we have cause for big blobs
+                // as args? This is the main driver of max possible input tx size
+                // at the moment, so making it smaller would reduce input tx size range
+                // significantly.
+                // Alternatively: could remove limits on pure argument size and number
+                // of publishes in a programmable tx, and replace with overall serialized
+                // size limits.
+                max_pure_argument_size: Some(100 * 1024),
```

Comment:
I would drop this to (10 * 1024). I think a 10k long string is reasonable and defensive enough.

Reply:
cool agreed